src/java.base/share/classes/sun/security/ssl/JsseJce.java

```diff
@@ -162,10 +188,38 @@ private static class EcAvailability {
         // Is EC crypto available?
         private static final boolean isAvailable;
 
+        /**
+         * Checks if a particular signature algorithm is available.
+         *
+         * @param algorithm The algorithm we will attempt to instantiate to check if it is available
+         * @return true if the signature algorithm is found, false otherwise
+         */
+        private static boolean isSignatureAlgorithmAvailable(String algorithm) {
+            try {
+                // Attempt to create a Cipher instance with the specified algorithm.
+                Signature.getInstance(algorithm);
+                return true;
+            } catch (NoSuchAlgorithmException e) {
```

Comment:
An earlier version of this also caught `NoSuchPaddingException` (which surprised me). Please confirm that we don't need to catch those exceptions.

Reply:
Yes I did compile this and a `NoSuchPaddingException` is not allowable by the compiler so I removed it.

Reply:
I just saw your comment https://github.com/ibmruntimes/openj9-openjdk-jdk/pull/705#discussion_r1414711935.